src/plugins/expression_reveal_image/public/components/reveal_image_component.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useRef, useState, useEffect, useCallback } from 'react';
+import { IInterpreterRenderHandlers } from 'src/plugins/expressions';
+import { NodeDimensions, RevealImageRendererConfig, OriginString } from '../../common/types';
+import { isValidUrl, elasticOutline } from '../../../presentation_util/public';
+import './reveal_image.scss';
+
+interface RevealImageComponentProps extends RevealImageRendererConfig {
+  handlers: IInterpreterRenderHandlers;
+  parentNode: HTMLElement;
+}
```

Comment:
For testability and concern separation maybe it would be useful to make this component more "presentational" and decouple it from the knowledge of "handlers".

Maybe `handlers` prop can be replaced by canonical React `onSomething` props:

```suggestion
interface RevealImageComponentProps extends RevealImageRendererConfig {
  onResize: () => void;
  onLoaded: () => void;
  onUnmount: () => void;
  parentNode: HTMLElement;
}
```

Reply:
I assume, that moving from `handlers` to `onSomething` will make developers, who will extract other plugins, write a lot of boilerplate code for covering this pattern. We already have a standardized way of providing listeners and emitters from the plugin, which is calling a renderer. It is describing types, possible events, etc. And current component is locked in a scope of a `renderer` and not going to be used separately.  I'm not really sure if it is needed.  
But at the same moment, I assume, that I don't see all the map and can be out of knowledge of the whole idea. I should refer to somebody, who is more knowledgeable.

@crob611, what do you think about it? Are there any reasons for making such changes?

Reply:
Ok, I'll provide such changes. Thanks for this suggestion.

Reply:
Done.